src/components/ActionableToast/ActionableToast.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { CSSProperties, useRef, useEffect, useCallback, useState } from 'react'
+import { Theme, useStyles } from '../../styles'
+import { Tooltip } from '../Tooltip'
+import { Button } from '../Button'
+import { Icon } from '../Icon'
+import { useLocale } from '../../i18n'
+
+export interface ActionableToastProps {
+  id: number
+  secondsVisible: number
+  title?: string
+  message: React.ReactNode
+  buttonLabel?: React.ReactNode
+  newToast?: boolean
+  action?: () => void
+  onClose?: () => void
+  removeToast: (id: number) => void
+}
+
+export function ActionableToast(props: ActionableToastProps) {
+  const {
+    id,
+    message,
+    onClose = () => {},
+    title,
+    buttonLabel = 'Button',
+    newToast,
+    removeToast,
+    action,
+    secondsVisible,
+  } = props
+
+  const locale = useLocale()
+  const timeoutRef = useRef<number | undefined>(undefined)
+  const { classes } = useStyles(createStyles)
+  const [isVisible, setIsVisible] = useState(false)
+
+  const handleCloseClick = () => {
+    onClose()
+    removeToast(id)
+  }
+
+  const clearTimer = () => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current)
+      timeoutRef.current = undefined
+    }
+  }
+
+  const startTimer = useCallback(() => {
+    timeoutRef.current = window.setTimeout(() => {
+      setIsVisible(false)
+      timeoutRef.current = window.setTimeout(() => {
```

Comment:
here u are setting `timeoutRef.current` again, if got canceled in those 500ms it will not clear the first `timeoutRef.current`, but i think its harmless, so u can let that way